src/Service/Validation/RegisterTokenValidator.cs

```diff
@@ -30,4 +31,16 @@ public static void ValidateAttestation(RegisterToken token, IFeaturesContext fea
             throw new ApiException("attestation_not_supported_on_plan", "Attestation type not supported on your plan", 400);
         }
     }
+
+    public static void Validate(this Token token, DateTimeOffset now)
```

Comment:
Passing `now` as a parameter seems anti-pattern. Maybe find a way to make it work by injecting TimeProvider.

Reply:
I should add the context that we do use Timeprovider, but since this is an extension method you can't inject, just use arguments. And it made less sense to send a timprovider as an argument compared to just a time (that itself is fetched from the timeprovider)

Reply:
But if that gets too complicated it would mean the structure of the code is often wrong elsewhere. You can even simplify the use of `TimeProvider` by injecting an interface and just abstract it.

Reply:
Each way has its trade-offs. 

By providing TimeProvider, we're controlling the inputs to the method (since its an abstraction around Time) and allows the method to encapsulate what a valid token is.

By providing `DateTimeOffset` called `now`, we're essentially doing the same thing except using the plain `DateTimeOffset` (which could be anything). 

TimeProvider kind of builds the "system" into the method and `now` conveys the intent...just allows any `DateTimeOffset` to be passed in.